### PR TITLE
Fix memory leak when prompt isn't explicitly dismissed

### PR DESCRIPTION
**Pull request by SUPERCILEX**
The animation listener doesn't die with the view apparently (which leaks it).

### CODE REVIEW

File: library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetPrompt.java [hunk 1]
```diff
@@ -887,6 +889,13 @@ else if (event.getAction() == KeyEvent.ACTION_UP
             return super.dispatchKeyEventPreIme(event);
         }
 
+        @Override
+        protected void onDetachedFromWindow()
+        {
+            super.onDetachedFromWindow();
+            mPrompt.cleanUpAnimation();
```
Comment (sjwall) [inline library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetPrompt.java]: Is there a reason that this isn't a call to `cleanUpPrompt`?

File: library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetPrompt.java [hunk 2]
```diff
@@ -188,6 +188,7 @@ public void run()
     {
         final ResourceFinder resourceFinder = promptOptions.getResourceFinder();
         mView = new PromptView(resourceFinder.getContext());
+        mView.mPrompt = this;
```
Comment (sjwall) [inline library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetPrompt.java]: Should this reference be set to null in `cleanUpPrompt`?

Comment (SUPERCILEX) [reply library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetPrompt.java]: We can, but it isn't necessary. The issue is the prompt holding on to the view, not the other way around.

File: library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetPrompt.java (hunk 1 above)
Comment (SUPERCILEX) [reply library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetPrompt.java]: `cleanUpPrompt` also notifies state changes which seems like we'd be introducing a race condition... I just realized I didn't share the leak with you, it's just the animator stuff which is leaking:
```
In com.supercilex.robotscouter.debug:2.3.1-DEBUG:1.
* com.supercilex.robotscouter.HomeActivity has leaked:
* GC ROOT android.widget.Toast$TN.mNextView
* references android.widget.LinearLayout.mParent
* references android.view.ViewRootImpl.mChoreographer
* references android.view.Choreographer.mCallbackQueues
* references array android.view.Choreographer$CallbackQueue[].[1]
* references android.view.Choreographer$CallbackQueue.mHead
* references android.view.Choreographer$CallbackRecord.action
* references android.animation.AnimationHandler$1.this$0 (anonymous implementation of android.view.Choreographer$FrameCallback)
* references android.animation.AnimationHandler.mAnimationCallbacks
* references java.util.ArrayList.elementData
* references array java.lang.Object[].[0]
* references android.animation.ValueAnimator.mUpdateListeners
* references java.util.ArrayList.elementData
* references array java.lang.Object[].[0]
* references uk.co.samuelwall.materialtaptargetprompt.MaterialTapTargetPrompt$10.this$0 (anonymous implementation of android.animation.ValueAnimator$AnimatorUpdateListener)
* references uk.co.samuelwall.materialtaptargetprompt.MaterialTapTargetPrompt.mView
* references uk.co.samuelwall.materialtaptargetprompt.MaterialTapTargetPrompt$PromptView.mContext
* leaks com.supercilex.robotscouter.HomeActivity instance

* Retaining: 6.5 kB.
* Reference Key: 8200773e-a087-4f54-8dd7-9fe208a43681
* Device: Google google Pixel 2 XL taimen
* Android Version: 8.1.0 API: 27 LeakCanary: 1.5.4 74837f0
* Durations: watch=5315ms, gc=172ms, heap dump=1438ms, analysis=154338ms
```

**Comment by sjwall** [comment]
Thanks again for your work and supplying the memory leak info

**Pull request merged**